feat(system): remove FlowTopologyHandler and TriggerEventPublisher

loicmathieu opened the pull request:
And make all flow modification pass throught the FlowService so that downstream consumers are always updated

loicmathieu committed:
feat(system): remove FlowTopologyHandler and TriggerEventPublisher

And make all flow modification pass throught the FlowService so that downstream consumers are always updated

fhussonnois reviewed (approved):
Cool, it will really make flow management easier if `FlowService` becomes the single point of entry for all operations. At some point, we should plan a refactoring of the `FlowController` that still contains too much business logic.

loicmathieu committed:
chore(system): split FlowService in two

- FlowValidationService with validation logic, not dependant of any repository so it can be used in CLI
- FlowService with now hard dependency on queue and repository

loicmathieu commented:
Most of the tests that fail here passed locally so I'll merge this one and see if after a rebase which fixes some issues in our CI pipeline the test still fails.
